apps/namada-interface/src/slices/notifications.ts

```diff
@@ -51,6 +60,22 @@ export const NotificationsSlice = createSlice({
       state.toasts = newToasts;
     },
   },
+  extraReducers: (builder) => {
+    builder.addMatcher(isAsyncThunkAction, (state, action) => {
```

Comment:
This one is pretty generic but it works quite well!
It matches `asyncThunkActions` and based on that either adds action as a pending one or removes it from state.
We  can then use it like that:
```ts
const isTransferSubmitting = useAppSelector<boolean>((state) =>
    state.notifications.pendingActions.includes(
      actionTypes.SUBMIT_TRANSFER_ACTION_TYPE
    )
  );
  ```

Reply:
That's nice!

Reply:
Looks good! nit: I guess there might be cases where we want to indicate something to be pending and reflect that in the UI with this mechanism, but not need a notification. In that sense having this under `state.notifications` might not be the perfect place, but that is no big deal to change at any time.

Reply:
Agree, we can move it at anytime though :)